protocol/bc/spend.go

```diff
@@ -12,6 +19,29 @@ type SpendInput struct {
 
 func (si *SpendInput) IsIssuance() bool { return false }
 
+func (si *SpendInput) writePrevoutCommitment(w io.Writer, assetVersion uint64) error {
+	return si.OutputCommitment.writeTo(w, assetVersion)
+}
+
+// readCommitment reads a spend input commitment AFTER the leading
+// type byte has been consumed.
+func (si *SpendInput) readCommitment(r io.Reader, txVersion, assetVersion uint64) (err error) {
```

Comment:
Do you mean you wish they had the same calling signature, maybe defined in an `interface`? The thing about that is, `IssuanceInput` commitments do not need the asset version right now, but `SpendInput` commitments do. We'd have to add unused parameters to the `IssuanceInput` version of `readCommitment`, which seems wrong in this case.

Would it be better if the two `readCommitment` functions had different names?